--- a/CPP/Windows/ErrorMsg.cpp  2020-05-28 00:35:02.729896917 +0200
+++ b/CPP/Windows/ErrorMsg.cpp  2020-05-28 00:40:01.676442629 +0200
@@ -13,7 +13,7 @@
   const char * txt = 0;
   AString msg;

-  switch(errorCode) {
+  switch(HRESULT(errorCode)) {
     case ERROR_NO_MORE_FILES   : txt = "No more files"; break ;
     case E_NOTIMPL             : txt = "E_NOTIMPL"; break ;
     case E_NOINTERFACE         : txt = "E_NOINTERFACE"; break ;
@@ -43,7 +43,7 @@
   const char * txt = 0;
   AString msg;

-  switch(messageID) {
+  switch(HRESULT(messageID)) {
     case ERROR_NO_MORE_FILES   : txt = "No more files"; break ;
     case E_NOTIMPL             : txt = "E_NOTIMPL"; break ;
     case E_NOINTERFACE         : txt = "E_NOINTERFACE"; break ;